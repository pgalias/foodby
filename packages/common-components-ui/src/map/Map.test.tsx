import React from 'react';
import { render } from '@testing-library/react';
import { Map, Marker } from '.';

describe('Map', () => {
  const location = {
    latitude: 54.414792427759004,
    longitude: 18.569411429474155,
  };

  test('should be a facade for the "outside" map component', () => {
    const Marker1 = () => <Marker center={location}>Foo</Marker>;
    const Marker2 = () => (
      <Marker
        center={{
          latitude: location.latitude + 0.02,
          longitude: location.longitude + 0.01,
        }}
      >
        Bar
      </Marker>
    );

    const { asFragment } = render(
      <Map center={location} zoom={13}>
        <Marker1 />
        <Marker2 />
      </Map>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
