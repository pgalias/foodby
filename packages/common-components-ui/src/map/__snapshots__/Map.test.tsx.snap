// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map should be a facade for the "outside" map component 1`] = `
<DocumentFragment>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://b.tile.openstreetmap.org/13/4518/2614.png"
              style="width: 256px; height: 256px; left: -143px; top: -41px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      >
        <img
          alt=""
          class="leaflet-marker-shadow leaflet-zoom-hide"
          src="marker-shadow.png"
          style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
        />
        <img
          alt=""
          class="leaflet-marker-shadow leaflet-zoom-hide"
          src="marker-shadow.png"
          style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 58px; top: -200px;"
        />
      </div>
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="marker-icon.png"
          style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="marker-icon.png"
          style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 58px; top: -200px; z-index: -200;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      >
        <div
          class="leaflet-control-zoom leaflet-bar leaflet-control"
        >
          <a
            aria-label="Zoom in"
            class="leaflet-control-zoom-in"
            href="#"
            role="button"
            title="Zoom in"
          >
            +
          </a>
          <a
            aria-label="Zoom out"
            class="leaflet-control-zoom-out"
            href="#"
            role="button"
            title="Zoom out"
          >
            −
          </a>
        </div>
      </div>
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="http://osm.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
